L1Trigger/Phase2L1ParticleFlow/interface/egamma/pftkegalgo_ref.h

```diff
@@ -0,0 +1,318 @@
+
+    void setDebug(int verbose) { debug_ = verbose; }
+
+    bool writeEgSta() const { return cfg.writeEgSta; }
+
+  private:
+    void link_emCalo2emCalo(const std::vector<EmCaloObjEmu> &emcalo, std::vector<int> &emCalo2emCalo) const;
+
+    void link_emCalo2tk(const PFRegionEmu &r,
+                        const std::vector<EmCaloObjEmu> &emcalo,
+                        const std::vector<TkObjEmu> &track,
+                        std::vector<int> &emCalo2tk) const;
+
+    //FIXME: still needed
+    float deltaPhi(float phi1, float phi2) const;
+
+    void sel_emCalo(unsigned int nmax_sel,
+                    const std::vector<EmCaloObjEmu> &emcalo,
+                    std::vector<EmCaloObjEmu> &emcalo_sel) const;
+
+    void eg_algo(const PFRegionEmu &region,
+                 const std::vector<EmCaloObjEmu> &emcalo,
+                 const std::vector<TkObjEmu> &track,
+                 const std::vector<int> &emCalo2emCalo,
+                 const std::vector<int> &emCalo2tk,
+                 std::vector<EGObjEmu> &egstas,
+                 std::vector<EGIsoObjEmu> &egobjs,
+                 std::vector<EGIsoEleObjEmu> &egeleobjs) const;
+
+    void addEgObjsToPF(std::vector<EGObjEmu> &egstas,
+                       std::vector<EGIsoObjEmu> &egobjs,
+                       std::vector<EGIsoEleObjEmu> &egeleobjs,
+                       const std::vector<EmCaloObjEmu> &emcalo,
+                       const std::vector<TkObjEmu> &track,
+                       const int calo_idx,
+                       const int hwQual,
+                       const pt_t ptCorr,
+                       const int tk_idx,
+                       const std::vector<unsigned int> &components = {}) const;
+
+    EGObjEmu &addEGStaToPF(std::vector<EGObjEmu> &egobjs,
+                           const EmCaloObjEmu &calo,
+                           const int hwQual,
+                           const pt_t ptCorr,
+                           const std::vector<unsigned int> &components) const;
+
+    EGIsoObjEmu &addEGIsoToPF(std::vector<EGIsoObjEmu> &egobjs,
+                              const EmCaloObjEmu &calo,
+                              const int hwQual,
+                              const pt_t ptCorr) const;
+
+    EGIsoEleObjEmu &addEGIsoEleToPF(std::vector<EGIsoEleObjEmu> &egobjs,
+                                    const EmCaloObjEmu &calo,
+                                    const TkObjEmu &track,
+                                    const int hwQual,
+                                    const pt_t ptCorr) const;
+
+    // FIXME: reimplemented from PFAlgoEmulatorBase
+    template <typename T>
+    void ptsort_ref(int nIn, int nOut, const std::vector<T> &in, std::vector<T> &out) const {
+      // // std based sort
+      // out = in;
+      // std::sort(out.begin(), out.end(),  std::greater<T>());
+      // out.resize(nOut);
+      out.resize(nOut);
+      for (int iout = 0; iout < nOut; ++iout) {
+        out[iout].clear();
+      }
```

Comment:
I think that for the structs that will enter as `T` the `0` initialization is already assured by the "default" constructor.

Reply:
I think we weren't sure that the default constructor of all the datamembers was zeroing things out.